internal/ethapi/api.go

```diff
@@ -714,8 +709,12 @@ type RPCTransaction struct {
 
 // newRPCPendingTransaction returns a pending transaction that will serialize to the RPC representation
 func newRPCPendingTransaction(tx *types.Transaction) *RPCTransaction {
-	from, _ := tx.FromFrontier()
-	v, r, s := tx.SignatureValues()
+	var signer types.Signer = types.FrontierSigner{}
+	if tx.Protected() {
+		signer = types.NewEIP155Signer(tx.ChainId())
+	}
+	from, _ := types.Sender(signer, tx)
+	v, r, s := types.SignatureValues(signer, tx)
```

Comment:
I think this should be `RawSignatureValues`. The v, r, s JSON fields are new in 1.5 and we can define the semantics for them. Raw values are preferable here because the client should be able to reconstruct the whole transaction.

e.g. if you do 

```go
client, _ := ethclient.Dial("http://127.0.0.1:8085")
tx, _ := client.TransactionByHash(ctx, common.HexToHash("...."))
from, _ := types.Sender(types.NewEIP155Signer(signer), tx))
```

The from value should be the same as it was on the server. Normalising the values breaks this.